### PR TITLE
Feat: Emulate Java Character class in actions & semantic predicates

### PR DESCRIPTION
**Pull request by BurtHarris**
This improves the portability of grammars which use the Java Character class to run in a ECMAScript/TypeScript environment by providing a Character class in the antlr4ts runtime and putting
it in scope in the action / semantic predicate generated code.   

Note: do not merge yet.

### CODE REVIEW

File: tool/src/org/antlr/v4/codegen/target/TypeScriptTarget.java [hunk 1]
```diff
@@ -142,6 +144,10 @@ protected STGroup loadTemplates() {
 		STGroup result = new STGroupFile(CodeGenerator.TEMPLATE_ROOT+"/TypeScript/"+getLanguage()+STGroup.GROUP_FILE_EXTENSION);
 		result.registerRenderer(Integer.class, new NumberRenderer());
 		result.registerRenderer(String.class, new StringRenderer());
+		result.registerRenderer(
+			org.antlr.v4.codegen.model.Action.class, 
+			new TypeScriptActionRenderer()
+			);
```
Comment (BurtHarris) [inline tool/src/org/antlr/v4/codegen/target/TypeScriptTarget.java]: @sharwell this does not seem to be working as I expected.   The TypeScriptActionRenderer is not getting invoked.   Can you help me understand why?

Comment (BurtHarris) [reply tool/src/org/antlr/v4/codegen/target/TypeScriptTarget.java]: @sharwell, this PR may have been overcome by events, but it showed me I don't the Java code for multiple target support.   Can you have a look and tell me why TypeScriptActionRenderer doesn't get invoked on this branch?

Comment (sharwell) [reply tool/src/org/antlr/v4/codegen/target/TypeScriptTarget.java]: I'm actually not sure. I've never tried to add a renderer for that type.

**Comment by BurtHarris** [comment]
I'll take this back and try to figure out...

**Comment by BurtHarris** [comment]
I'm so out-of-context on this pull request, its probably best to close it for now.

**Comment by BurtHarris** [comment]
@sharwell, Perhaps it makes sense for us to make a selected set of examples ported to antlr4ts.   I had done some work using gulp on a build system change that could make it simpler to do this inside this repository, but the changes kept spinning wildly out-of-control and I didn't want to submit them.    I learned a good bit about gulp in the process and might want to give it another stab, but I want to check with you to avoid doing it when you are on a roll (as you seem to be recently :-).   Let me know what you think.

**Comment by sharwell** [comment]
I'm not having a particular problem with NPM right now. Given limited time I think I'd prefer to not go learning about something new and (possibly) totally different.

**Pull request closed without merging**